core/daemon/src/main/java/org/opennms/netmgt/vmmgr/Controller.java

```diff
@@ -192,7 +192,12 @@ static void interpolateSystemProperties() {
         final Scope scope = new SecureCredentialsVaultScope(secureCredentialsVault);
 
         for(final Map.Entry<Object, Object> entry : System.getProperties().entrySet()) {
-            System.setProperty(entry.getKey().toString(), Interpolator.interpolate(entry.getValue().toString(), scope).output);
+            if (entry.getKey() == null || entry.getValue() == null || !(entry.getValue() instanceof String)) {
+                continue;
+            }
+            if (((String) entry.getValue()).contains("${scv:")) {
```

Comment:
Let's add a specific comment here

// Since we are only interpolating scv related properties, we restrict this to interpolating only scv related properties.